model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -189,9 +198,24 @@ object AddressResponseAddress {
       northing <- Try(other.northing.toInt)
     } yield AddressResponseGeo(latitude, longitude, easting, northing)
 
+    val saoLeftRangeExists = other.saoStartNumber != "" || other.saoStartSuffix != ""
+    val saoRightRangeExists = other.saoEndNumber != "" || other.saoEndSuffix != ""
+    val saoHyphen = if (saoLeftRangeExists && saoRightRangeExists) "-" else ""
+    val saoNumbers = Seq(other.saoStartNumber, other.saoStartSuffix, saoHyphen, other.saoEndNumber, other.saoEndSuffix).mkString("")
```

Comment:
`s"${var.x} ${var.y} ${var.q}".split(" ").mkString(" ")` is harder to read and understand (for me) than
`Seq(var.x, var.y, var.q).filter(_.nonEmpty).mkString(" ")` I already told you so